radio/src/lua/api_model.cpp

```diff
@@ -1586,11 +1586,69 @@ static int luaModelResetSensor(lua_State *L)
   if (idx < MAX_TELEMETRY_SENSORS) {
     telemetryItems[idx].clear();
   }
-  
+
   lua_pushnil(L);
   return 1;
 }
+/*luadoc
+@function model.swashRingData(index, line,{parameters})
+
+Set heli swash parameters
+
+index = 0
+line = 0
+@param (table):
+* `type` (number in quotes) "1"=120, "2"=120X, "3"=140, "4"=90
+* `value` (number) swash ring value (normally 0)
+* 'collectiveSource' (number) source index
+* 'aileronSource' (number) source index
+* 'elevatorSource' (number) source index
+* 'collectiveWeight'(value) -100 to 100
+* 'aileronWeight' (value) -100 to 100
+* 'elevatorWeight' (value) -100 to 100
+
+* @notice If a parameter is missing, then that parameter remains unchanged.
 
+@status current Introduced in 2.3.x
+*/
+static int luaModelSwashRingData(lua_State *L)
+{
+unsigned int idx = luaL_checkunsigned(L, 1);
+if (idx < MIXSRC_CYC3){
```

Comment:
You seemt to be checking idx and not using it ?

Reply:
unused idx removed,
it was added to reflect the other lua set commands format.
I didn't think this would work without an initial "if" statement ... I stand corrected. 

That's what I get for not coding C in over 35 years... second guessing myself and adding unnecessary items